### ISSUE TITLE
Rule S4143: False positive when when other object with same Dictionary property name exists

### ISSUE THREAD

**Issue by marcells** [SonarSource/sonar-dotnet#2160]
### Description & Repro steps

In the following code snippet the rule should not be violated, because both dictionaries are different.
```
[...]
_contactFolder.Contacts[index] = contact;
Contacts[index] = contact;
[...]
```

If possible then please create a git repository with a repro sample or attach a zip to the issue.

### Expected behavior

Rule should not be violated.

### Actual behavior

Rule is violated.

### Related information

* SonarC# Version: 4.5.1
* Visual Studio Version: 15.9.4

Running in TFS. Extension installed via the TFS Marektplace.

**Comment by valhristov** [comment]
Duplicate of #1897
